shigatyper/2.0.1/Dockerfile

```diff
@@ -0,0 +1,108 @@
+# FROM defines the base docker image. This command has to come first in the file
+# The 'as' keyword lets you name the folowing stage. We use `app` for the production image
+FROM ubuntu:xenial as app
+
+# ARG sets environment variables during the build stage
+# USHER ver is in the conda version style.
+ARG SHIGATYPER_VER="2.0.1"
+ARG SAMTOOLSVER="1.15"
+ARG MINIMAP2_VER="2.24"
+ARG bcftoolsVer="1.14"
+
+
+# LABEL instructions tag the image with metadata that might be important to the user
+# Optional, but highly recommended
+LABEL base.image="ubuntu:xenial"
+LABEL dockerfile.version="1"
+LABEL software="shigatyper"
+LABEL software.version=$SHIGATYPER_VER
+LABEL description="Determine Shigella serotype using Illumina (single or paired-end) or Oxford Nanopore reads!"
+LABEL website="https://github.com/StaPH-B/docker-builds"
+LABEL license="https://github.com/StaPH-B/docker-builds/blob/master/LICENSE"
+LABEL maintainer="John Arnn"
+LABEL maintainer.email="jarnn@utah.gov"
+
+# RUN executes code during the build
+# Install dependencies via apt-get or yum if using a centos or fedora base
+RUN apt-get update && apt-get install -y \
+ wget \
+ git \
+ libncurses5-dev \
+ libbz2-dev \
+ liblzma-dev \
+ libcurl4-gnutls-dev \
+ zlib1g-dev \
+ libssl-dev \
+ gcc \
+ make \
+ perl \
+ bzip2 \
+ gnuplot \
+ ca-certificates \
+ gawk \
+ curl \
+ build-essential && \
+ apt-get autoclean && rm -rf /var/lib/apt/lists/*
+
+ # shigatyper depends on samtools
+RUN wget https://github.com/samtools/samtools/releases/download/${SAMTOOLSVER}/samtools-${SAMTOOLSVER}.tar.bz2 && \
+   tar -xjf samtools-${SAMTOOLSVER}.tar.bz2 && \
+   rm samtools-${SAMTOOLSVER}.tar.bz2 && \
+   cd samtools-${SAMTOOLSVER} && \
+   ./configure && \
+   make && \
+   make install && \
+   cd ..
+
+ # shigatyper depends on MINIMAP2
+RUN curl -L https://github.com/lh3/minimap2/releases/download/v${MINIMAP2_VER}/minimap2-${MINIMAP2_VER}_x64-linux.tar.bz2 | tar -jxvf -
+ENV PATH="${PATH}:/minimap2-${MINIMAP2_VER}_x64-linux"
+
+ # shigatyper depends on bcftools
+RUN wget https://github.com/samtools/bcftools/releases/download/${bcftoolsVer}/bcftools-${bcftoolsVer}.tar.bz2 && \
+  tar -vxjf bcftools-${bcftoolsVer}.tar.bz2 && \
+  rm bcftools-${bcftoolsVer}.tar.bz2 && \
+  cd bcftools-${bcftoolsVer} && \
+  make && \
+  make install && \
+  cd ..
+
+# get miniconda and the shigatyper repo
+RUN wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh && \
```

Comment:
It looks like shigatyper is relying on conda to package it. Should we attempt to install it directly from a github download?

Reply:
The only installation offered on the github page is a conda package. But let me see if I can use that package without using conda.

Reply:
So I am running into some versioning problems trying to avoid conda. The repository that apt-get install uses are older versions of python, 3.5.2,  and pip. Without updating pip I get this error:
UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 5537: ordinal not in range(128)
And also a line that suggests an update for pip. If I then put in a line that updates pip I then get this error: 
 File "/usr/local/lib/python3.5/dist-packages/pip/_internal/cli/main.py", line 57
    sys.stderr.write(f"ERROR: {exc}")
Which I believe is caused by the pip version is for newer releases of python and no longer works with this version of python. 

I have not found a way to get a version of python that might work. It might be just as dependable to use bioconda for this tool.

Reply:
Good to know!